### DISCUSSION TITLE
What happens when an auth collection reaches 99999 number on a username?

### DISCUSSION THREAD

**Discussion by pagoru** [pocketbase/pocketbase#4570] [Q&A] [answered]
What happens when an auth collection reaches 99999 number on a username? It goes to 6 numbers or what happens?
I'm currently not using the username, so I'm just curious

**Comment by ganigeorgiev** [comment, marked as answer]
If the autogenerated username exists we'll try again (10x times max) by appending another random number. In the worst case scenario (it is extremely unlikely), aka. all autogenerated usernames exists, the user operation will just fail with "username not unique error".

_Note that the `username` field will be removed after the [refactoring](https://github.com/pocketbase/pocketbase/discussions/4355) and users will have the option to define their own `username` field if they need it._

**Reply by pagoru** [reply]
Really nice, thanks! 👯
